src/Microsoft.OpenApi/Services/OpenApiFilterService.cs

```diff
@@ -46,16 +56,52 @@ public static class OpenApiFilterService
                 {
                     var regex = new Regex(tagsArray[0]);
 
-                    predicate = (o) => o.Tags.Any(t => regex.IsMatch(t.Name));
+                    predicate = (url, operationType, operation) => operation.Tags.Any(tag => regex.IsMatch(tag.Name));
                 }
                 else
                 {
-                    predicate = (o) => o.Tags.Any(t => tagsArray.Contains(t.Name));
+                    predicate = (url, operationType, operation) => operation.Tags.Any(tag => tagsArray.Contains(tag.Name));
+                }
+            }
+            else if (requestUrls != null)
+            {
+                List<string> operationTypes = new List<string>();
+
+                var graphVersion = source.Info.Version;
+
+                var sources = new Dictionary<string, OpenApiDocument> {{graphVersion, source}};
+                var rootNode = CreateOpenApiUrlTreeNode(sources);
+
+                //Iterate through urls dictionary and fetch operations for each url
+                foreach (var path in requestUrls)
+                {
+                    var serverList = source.Servers;
+                    var url = FormatUrlString(path.Key, serverList);
+
+                    var openApiOperations = GetOpenApiOperations(rootNode, url, graphVersion);
+                    if (openApiOperations == null)
+                    {
+                        continue;
+                    }
+
+                    foreach (var ops in openApiOperations)
+                    {
+                        operationTypes.Add(ops.Key + url);
+                    }
+                }
+
+                if (!(bool)operationTypes?.Any())
```

Comment:
You really don't need to check for nullability, `operationTypes` will never be null as it is initialized during declaration.
```suggestion
                if (!operationTypes.Any())
```